cognite/extractorutils/uploader.py

```diff
@@ -512,6 +524,37 @@ def __init__(
         self.latency = TIMESERIES_UPLOADER_LATENCY
         self.latency_zero_point = arrow.utcnow()
 
+    def _verify_datapoint_time(self, time: Union[int, float, datetime]) -> bool:
+        if isinstance(time, int) or isinstance(time, float):
+            return not math.isnan(time) and time >= MIN_DATAPOINT_TIMESTAMP
+        else:
+            return time.timestamp() * 1000.0 >= MIN_DATAPOINT_TIMESTAMP
+
+    def _verify_datapoint_value(self, value: Union[int, float, str]) -> bool:
+        if isinstance(value, float):
+            return not (
+                math.isnan(value) or math.isinf(value) or value > MAX_DATAPOINT_VALUE or value < MIN_DATAPOINT_VALUE
+            )
+        elif isinstance(value, str):
+            return len(value) <= MAX_DATAPOINT_STRING_LENGTH
+        else:
+            return True
+
+    def _is_datapoint_valid(
+        self,
+        dp: Union[
+            Dict[Union[int, float, datetime], Union[int, float, str]],
+            Tuple[Union[int, float, datetime], Union[int, float, str]],
+        ],
+    ) -> bool:
+        if isinstance(dp, Dict):
+            return self._verify_datapoint_time(dp["timestamp"]) and self._verify_datapoint_value(dp["value"])
+        elif isinstance(dp, Tuple):
+            return self._verify_datapoint_time(dp[0]) and self._verify_datapoint_value(dp[1])
+        else:
+            # I don't know how many different legal permutations there are, since the type hints are incomplete.
```

Comment:
I'm a bit unsure what you mean by incomplete type hints? If the user sends arguments that doesn't match the type hints, that's on them, and I think just returning here (like we are) is fine.

Reply:
You are right, of course. I wrote this with a slight misunderstanding, the type hints for DataPointList are actually wrong, from the SDK.